server/api.js

```diff
@@ -211,7 +165,34 @@ router.post("/laptop_donation", (req, res) => {
 		email,
 		deliveryOption,
 	])
-		.then(() => res.send("result.rows"))
+		.then(async () => {
+			let laptopDonationResult = await db.query(
+				"SELECT * FROM laptop_donation d WHERE (SELECT COUNT(*) FROM laptop_assignment a WHERE a.laptop_donation_id = d.id) < d.number_of_laptops ORDER BY d.id LIMIT 1"
+			);
+			let laptopDonation = {};
+			if (laptopDonationResult.rows.length > 0) {
+				laptopDonation = {
+					id: laptopDonationResult.rows[0].id,
+				};
+			}
+			if (laptopDonation.id) {
+				const requestIDResult = await db.query(
+					"SELECT id FROM laptop_request where id is not null AND id NOT IN (SELECT laptop_request_id FROM laptop_assignment)"
+				);
+				let requestID = {};
```

Comment:
requestId, not requestId, but why is it an object and not just an id?

Reply:
using just the id